media-autobuild_suite.bat

```diff
@@ -1354,142 +1337,156 @@ if not exist %instdir%\mintty.lnk (
+    pacman -S --needed --ask=20 --noconfirm --asdeps bash pacman msys2-runtime
 
     echo.-------------------------------------------------------------------------------
     echo.second update
     echo.-------------------------------------------------------------------------------
-    (
-        echo.echo -ne "\033]0;second msys2 update\007"
-        echo.pacman --noconfirm -Syu --asdeps
-        echo.exit
-    )>%build%\secondUpdate.sh
-    call :runBash secondUpdate.log "%build%\secondUpdate.sh"
-    del %build%\secondUpdate.sh
+    title second msys2 update
+    call :runBash secondUpdate.log pacman --noconfirm -Syu --asdeps
 
     (
         echo.Set Shell = CreateObject^("WScript.Shell"^)
         echo.Set link = Shell.CreateShortcut^("%instdir%\mintty.lnk"^)
         echo.link.Arguments = "-full-path -mingw"
         echo.link.Description = "msys2 shell console"
         echo.link.TargetPath = "%instdir%\%msys2%\msys2_shell.cmd"
-        echo.link.WindowStyle = ^1
+        echo.link.WindowStyle = 1
         echo.link.IconLocation = "%instdir%\%msys2%\msys2.ico"
         echo.link.WorkingDirectory = "%instdir%\%msys2%"
         echo.link.Save
     )>%build%\setlink.vbs
-    cscript /nologo %build%\setlink.vbs
+    cscript /B /Nologo %build%\setlink.vbs
     del %build%\setlink.vbs
 )
 
+rem createFolders
+if %build32%==yes call :createBaseFolders local32
+if %build64%==yes call :createBaseFolders local64
+
+rem checkFstab
+set "removefstab=no"
+set "fstab=%instdir%\%msys2%\etc\fstab"
+if exist %fstab%. (
+    findstr build32 %fstab% >nul 2>&1 && set "removefstab=yes"
+    findstr trunk %fstab% >nul 2>&1 || set "removefstab=yes"
+    for /f "tokens=1 delims= " %%a in ('findstr trunk %fstab%') do if not [%%a]==[%instdir%\] set "removefstab=yes"
+    findstr local32 %fstab% >nul 2>&1 && ( if [%build32%]==[no] set "removefstab=yes" ) || if [%build32%]==[yes] set "removefstab=yes"
+    findstr local64 %fstab% >nul 2>&1 && ( if [%build64%]==[no] set "removefstab=yes" ) || if [%build64%]==[yes] set "removefstab=yes"
+)
+
+if not [%removefstab%]==[no] (
+    rem writeFstab
+    echo -------------------------------------------------------------------------------
+    echo.
+    echo.- write fstab mount file
+    echo.
+    echo -------------------------------------------------------------------------------
+    (
+        echo.none / cygdrive binary,posix=0,noacl,user 0 0
+        echo.
+        echo.%instdir%\ /trunk
+        echo.%instdir%\build\ /build
+        echo.%instdir%\%msys2%\mingw32\ /mingw32
+        echo.%instdir%\%msys2%\mingw64\ /mingw64
+        if "%build32%"=="yes" echo.%instdir%\local32\ /local32
+        if "%build64%"=="yes" echo.%instdir%\local64\ /local64
+    )>"%instdir%\%msys2%\etc\fstab."
+)
+
 if not exist "%instdir%\%msys2%\home\%USERNAME%" mkdir "%instdir%\%msys2%\home\%USERNAME%"
 set "TERM="
+type nul >>"%instdir%\%msys2%\home\%USERNAME%\.minttyrc"
```

Comment:
Since you overwrite this file every time, the check below is redundant.

Reply:
.minttyrc is not actually overwritten, rather `type nul >> file` is like touch file as it will create if it doesn't exist, but nothing else

Reply:
https://ss64.com/nt/syntax-redirection.html
I see.

Reply:
I did not realize that batch supports the `<` redirect. That's actually good to know